ibm/service/power/resource_ibm_pi_instance_snapshot.go

```diff
@@ -0,0 +1,288 @@
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID, snapshotID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	snapshot := instance.NewIBMPISnapshotClient(ctx, sess, cloudInstanceID)
+	snapshotdata, err := snapshot.Get(snapshotID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set(Arg_SnapshotName, snapshotdata.Name)
+	d.Set(Attr_CreationDate, snapshotdata.CreationDate.String())
+	d.Set(Attr_LastUpdateDate, snapshotdata.LastUpdateDate.String())
+	d.Set(Attr_SnapshotID, *snapshotdata.SnapshotID)
+	d.Set(Attr_Status, snapshotdata.Status)
+	d.Set(Attr_VolumeSnapshots, snapshotdata.VolumeSnapshots)
+
+	return nil
+}
+
+func resourceIBMPIInstanceSnapshotUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID, snapshotID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	client := instance.NewIBMPISnapshotClient(ctx, sess, cloudInstanceID)
+
+	if d.HasChange(Arg_SnapshotName) || d.HasChange(Arg_Description) {
+		name := d.Get(Arg_SnapshotName).(string)
+		description := d.Get(Arg_Description).(string)
+		snapshotBody := &models.SnapshotUpdate{Name: name, Description: description}
+
+		_, err := client.Update(snapshotID, snapshotBody)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		_, err = isWaitForPIInstanceSnapshotAvailable(ctx, client, snapshotID, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceIBMPISnapshotRead(ctx, d, meta)
+}
+
+func resourceIBMPIInstanceSnapshotDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID, snapshotID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	client := instance.NewIBMPISnapshotClient(ctx, sess, cloudInstanceID)
+	snapshot, err := client.Get(snapshotID)
+	if err != nil {
+		// snapshot does not exist
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("The snapshot  to be deleted is in the following state .. %s", snapshot.Status)
+
+	err = client.Delete(snapshotID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = isWaitForPIInstanceSnapshotDeleted(ctx, client, snapshotID, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func isWaitForPIInstanceSnapshotAvailable(ctx context.Context, client *instance.IBMPISnapshotClient, id string, timeout time.Duration) (interface{}, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_InProgress},
+		Target:     []string{State_Available, State_ACTIVE},
```

Comment:
The refresh func is never returning `State_ACTIVE`, can add "" to the target list.